src/main/java/com/redhat/pantheon/servlet/GetObjectByUUID.java

```diff
@@ -0,0 +1,79 @@
+package com.redhat.pantheon.servlet;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestDispatcherOptions;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletPaths;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import javax.jcr.ItemNotFoundException;
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.redhat.pantheon.servlet.ServletUtils.*;
+
+/**
+ * Simple servlet which aims to provide an internal way to quickly access resources by their UUID. This servlet is
+ * read-only, and the returned content is in json format. This servlet also offers the capability of deep diving into
+ * the resource's children by providing a 'depth' parameter.
+ *
+ * @author Carlos Munoz
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet which allows querying of any object via their UUID",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/pantheon/internal/object.json")
```

Comment:
instead of "/pantheon/internal/object.json", I prefer  "/pantheon/internal/node.json". This object is essentially a JCR node.